tajo-core/src/main/java/org/apache/tajo/engine/planner/physical/HashLeftOuterJoinExec.java

```diff
@@ -211,12 +169,16 @@ protected void loadRightToHashTable() throws IOException {
       }
 
       List<Tuple> newValue = tupleSlots.get(keyTuple);
-      if (newValue != null) {
-        newValue.add(tuple);
-      } else {
-        newValue = new ArrayList<Tuple>();
-        newValue.add(tuple);
-        tupleSlots.put(keyTuple, newValue);
+      try {
+        if (newValue != null) {
+          newValue.add(tuple.clone());
```

Comment:
This problem is quite complicated. So, I create a new issue (https://issues.apache.org/jira/browse/TAJO-1343) to handle this problem.
Please review the patch.